monitoring/iscsi.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package monitoring
+
+import (
+	"context"
+
+	"github.com/gravitational/satellite/agent/health"
+	pb "github.com/gravitational/satellite/agent/proto/agentpb"
+
+	"github.com/coreos/go-systemd/v22/dbus"
+	"github.com/gravitational/trace"
+)
+
+const (
+	iscsiCheckerID = "iscsi"
+
+	// ISCSIDService is the name of the iscsid systemd service unit.
+	ISCSIDService = "iscsid.service"
+	// ISCSIDSocket is the name of the iscsid systemd socket unit.
+	ISCSIDSocket = "iscsid.socket"
+)
+
+type FormatISCSIError func(unitName string) string
+
+// NewISCSIChecker verifies that the iscsid service is not running on the host.
+// The fmt parameter formats the message shown to the user when the probe fails.
+func NewISCSIChecker(fmt FormatISCSIError) health.Checker {
+	return &iscsiChecker{FailedProbeMsgFmt: fmt}
+}
+
+type iscsiChecker struct {
+	FailedProbeMsgFmt FormatISCSIError
+}
+
+// Name returns the name of this checker
+// Implements health.Checker
+func (c iscsiChecker) Name() string {
+	return iscsiCheckerID
+}
+
+// Check will check the systemd unit data coming from dbus to verify that
+// there are no iscsid services present on the host.
+func (c iscsiChecker) Check(ctx context.Context, reporter health.Reporter) {
+	conn, err := dbus.New()
```

Comment:
Recommend just logging dbus connection error as it does not necessarily verify iscsid service status. Same below with `conn.ListUnits()` error.

I think a false positive is better than a false negative here since this checker reports a failed probe and will degrade the cluster.

Reply:
I think that this is correct behavior. If we cannot query the status of the systemd units via dbus then this means that something major is not ok with the system and the cluster deserves to be in a degraded state. 

We cannot say that the cluster is ok if we cannot query the state of the systemd units via dbus.  Dbus and systemd are very important components and if we cannot query them then we should definitely declare the cluster as degraded.